File: frontend/src/types/book.ts
export interface NewBook {
  title: string
  author: string
  isbn: string
  price: number
  publisher?: string
}

export interface Book extends NewBook {
  id: number
}

export interface BookPage {
  content: Book[]
  totalElements: number
  totalPages: number
  size: number
  number: number
}

File: backend/src/main/kotlin/org/example/entity/Book.kt
package org.example.entity

import jakarta.persistence.*

@Entity
@Table(name = "books")
data class Book(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false)
    val title: String,

    @Column(nullable = false)
    val author: String,

    @Column(nullable = false, unique = true)
    val isbn: String,

    @Column(nullable = false)
    val price: Double,

    @Column
    val publisher: String? = null
) {
    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "book_authors",
        joinColumns = [JoinColumn(name = "book_id")],
        inverseJoinColumns = [JoinColumn(name = "author_id")]
    )
    var authors: MutableSet<Author> = mutableSetOf()
}
